### PR TITLE
Add sleep before looking for dump

### PR DESCRIPTION
**Pull request by danmoseley**
Relates to https://github.com/dotnet/core-eng/issues/10238

### CODE REVIEW

**Review by stephentoub — APPROVED** [review]

File: eng/testing/RunnerTemplate.sh [hunk 1]
```diff
@@ -170,7 +170,12 @@ fi
 # ======================= BEGIN Core File Inspection =========================
 pushd $EXECUTION_DIR >/dev/null
 if [[ "$(uname -s)" == "Linux" && $test_exitcode -ne 0 ]]; then
-  echo Looking around for any Linux dump...
+  have_sleep=$(which sleep)
+  if [ -x "$have_sleep" ] ; then
```
Comment (safern) [inline eng/testing/RunnerTemplate.sh:174]: Should we do this only on helix machines? Maybe checking a helix env var?

**Review by ViktorHofer — COMMENTED** [review]
Presumably we won't need that with dotnet test for which we don't use the run template so LGTM.

File: eng/testing/RunnerTemplate.sh (hunk 1 above)
Comment (danmosemsft) [reply eng/testing/RunnerTemplate.sh:174]: oh! How did I miss that.

Comment (danmosemsft) [reply eng/testing/RunnerTemplate.sh]: look correct now? I tested locally.

**Review by safern — APPROVED** [review]

**Comment by danmosemsft** [comment]
@DrewScoggins the dotnet-runtime-perf job failed but the only clue I see is the last line, "Killed". I can't find any other logs.

https://helix.dot.net/api/2019-06-17/jobs/851c5dd4-168c-40ba-9910-eff5886e34b1/workitems/release.Partition14/console

**Comment by ViktorHofer** [comment]
Would a better place to do this be here https://github.com/dotnet/runtime/blob/master/src/libraries/sendtohelixhelp.proj#L139?

**Comment by danmosemsft** [comment]
@ViktorHofer it has to happen between the tests executing and the searching for dumps, those both happen in this file right?

**Comment by safern** [comment]
>@ViktorHofer it has to happen between the tests executing and the searching for dumps, those both happen in this file right?

Yeah. We could use `HelixPostCommands` but that seems more hacky. I rather have all commands in the same place, it would be really hard to find where all this is happening if we have it split in between multiple places.

**Comment by danmosemsft** [comment]
OK based on the above, merging and let's see/hope we start getting dumps more reliably. LMK if you see a segfault/assert failure on Linux that doesn't get a dump..

**Pull request merged**